storehaus-kafka/src/main/scala/com/twitter/storehaus/kafka/KafkaSink.scala

```diff
@@ -78,40 +74,36 @@ object KafkaSink {
   type Dispatcher[K, V] = ((K, V)) => Future[Unit]
 
   /**
-   * Creates KafkaSink by wrapping KafkaStore
-   * @param store KafkaStore
-   * @tparam K key
-   * @tparam V value
-   * @return KafkaSink
-   */
-  def apply[K, V](store: KafkaStore[K, V]): KafkaSink[K, V] = {
-    val sink = new KafkaSink[K, V](store.put)
-    sink
-  }
+    * Create a KafkaSink by wrapping a KafkaStore
+    * @param store KafkaStore
+    * @return KafkaSink[K, V]
+    */
+  def apply[K, V](store: KafkaStore[K, V]): KafkaSink[K, V] =
+    new KafkaSink[K, V](store.put)
 
   /**
-   * Returns KafkaSink[K,V]
-   * @param brokers  kafka brokers
-   * @param topic kafka topic.
-   * @tparam K key
-   * @tparam V value
-   * @return KafkaSink[K,V]
-   */
-  def apply[K, V, E <: Encoder[V] : Manifest](brokers: Seq[String], topic: String): KafkaSink[K, V] = {
-    val store = KafkaStore[K, V, E](brokers, topic)
+    * Create a KafkaSink
+    * @param topic Kafka topic
+    * @param brokers Addresses of the Kafka brokers in the hostname:port format
+    * @return KafkaSink[K, V]
+    */
+  def apply[K, V, KS <: Serializer[K] : Manifest, VS <: Serializer[V] : Manifest](
+    topic: String,
+    brokers: Seq[String]
+  ): KafkaSink[K, V] = {
+    val store = KafkaStore[K, V, KS, VS](topic, brokers)
     lazy val sink = apply[K, V](store)
     sink
```

Comment:
no need for this lazy here... is there? Looks like a misunderstanding of how lazy works.